mozjs-sys/build.rs

```diff
@@ -57,9 +57,20 @@ fn main() {
     // Used by mozjs downstream, don't remove.
     println!("cargo:outdir={}", build_dir.display());
 
-    if let Some(archive) = env::var_os("MOZJS_ARCHIVE") {
-        download_static_lib_binaries(&PathBuf::from(archive), &build_dir);
-    } else {
+    // Link to pre-built archive first if it exists.
+    let build_from_source = match env::var_os("MOZJS_FROM_SOURCE") {
+        Some(_) => true,
+        None => match link_static_lib_binaries(&build_dir) {
+            Ok(()) => false,
+            Err(e) => {
+                println!("cargo:warning=Failed to link pre-built archive by {e}. Building from source instead.");
+                true
+            }
+        },
+    };
```

Comment:
I updated the step of linking prebuilt archive to return Result. If it fails, it will try to build it from source instead.
This means if the target isn't correct or the path `MOZJS_ARCHIVE` doesn't exist, it will continue to build from source.

Reply:
I would just make that we fallback if automatic fails, so if one used `MOZJS_ARCHIVE`, they wanted something explicitly so we should hard fail to let them know something is wrong.

Reply:
Also I think we should only fallback if download or extraction fail, if there is any other problem we should hard fail.